src/Modules/OrchardCore.Commerce/Abstractions/IProductService.cs

```diff
@@ -29,9 +29,11 @@ public interface IProductService
 
 public static class ProductServiceExtensions
 {
+    public static int GetDashIndex(string sku) => sku.IndexOf(value: "-", StringComparison.InvariantCulture);
```

Comment:
This is not an extension method.

Do it like this:
```suggestion
    public static int GetDashIndex(this string sku) => sku.IndexOf(value: "-", StringComparison.InvariantCulture);
```
Then you can use it like this
```
var dashIndex = sku.GetDashIndex();
```

Also could you please move the whole `ProductServiceExtensions` class to a different file to the `Extensions` folder. It's getting ugly that it's here.

Reply:
The `comparisonType` should be `StringComparison.Ordinal`. When you compare punctuation this is the simplest and preferred type.
Also I don't think this needs to be an extension method because using split would be better in both use cases.